loguru.hpp

```diff
@@ -102,6 +102,35 @@ Website: www.ilikebigbits.com
 #include <sal.h>	// Needed for _In_z_ etc annotations
 #endif
 
+/*
+	Will add syslog as a standard sink for log messages
+	Any logging message with a verbosity lower or equal to
+	the given verbosity will be included.
+
+	This works for Unix like systems (i.e. Linux/Mac)
+	There is no current implementation for Windows (as I don't know the
+	equivalent calls or have a way to test them). If you know please
+	add and send a pull request.
+
+	The code should still compile under windows but will only generate
+	a warning message that syslog is unavailable.
+
+	Search for LOGURU_SYSLOG to find and fix.
+*/
+#if defined(__linux__) || defined(__APPLE__)
+#define LOGURU_SYSLOG 1
+#endif
+
+#ifdef LOGURU_SYSLOG
+// This is required here as LOG_USER is used as a default value to add_syslog()
+#include <syslog.h>
+#else
+/* To make the code compile on systems with syslog define LOG_USER
+   so it can be used as default parameter.
+*/
+#define LOG_USER 0
+#endif
```

Comment:
This whole section should be removed, as I said I don't want any `#include`:s in `loguru.hpp`.

Here is one way to solve it:

``` C++
// Send logs to syslog with LOG_USER facility
bool add_syslog(const char* app_name, Verbosity verbosity);

// Send logs to syslog with your own choice of facility (LOG_USER, LOG_AUTH, ...)
bool add_syslog(const char* app_name, Verbosity verbosity, int facility);
```

Reply:
OK